src/Exception/HttpException.php

```diff
@@ -0,0 +1,84 @@
+<?php
+
+namespace Http\Client\Exception;
+
+use Psr\Http\Message\RequestInterface;
+use Psr\Http\Message\ResponseInterface;
+
+/**
+ * Thrown when both a request and a response are available
```

Comment:
or maybe "Thrown when a response was received but has an error status code. This exception always provides the request and response objects."